tests/Unit/Domain/CoordinateMaps/TimeDependent/Test_Shape.cpp

```diff
@@ -0,0 +1,442 @@
+                       std::unique_ptr<FunctionsOfTime::FunctionOfTime>>;
+
+std::vector<std::vector<std::complex<double>>> generate_random_coefs(
+    size_t l_max, size_t m_max, gsl::not_null<std::mt19937*> generator) {
+  std::vector<std::vector<std::complex<double>>> coefs{};
+  // if the coefficients are made too large, the map has a good chance of
+  // mapping through the center, causing the test to fail
+  std::uniform_real_distribution<double> coef_dis{0.1, 0.2};
+  const auto ran = [&coef_dis, &generator]() { return coef_dis(*generator); };
+
+  for (size_t l = 0; l <= l_max; ++l) {
+    std::vector<std::complex<double>> tmp{ran()};
+    for (size_t m = 1; m <= std::min(l, m_max); ++m) {
+      tmp.emplace_back(ran() + ran() * imag);
+    }
+    coefs.emplace_back(tmp);
+  }
+  // set monopole and dipole to zero
+  coefs.at(0).at(0) = 0.;
+  coefs.at(1).at(0) = 0.;
+  coefs.at(1).at(1) = 0.;
+  return coefs;
+}
+
+// converts complex coefficients to format expected by spherepack (eq. 25 of
+// spherical harmonics notes)
+DataVector convert_coefs_to_spherepack(
+    const std::vector<std::vector<std::complex<double>>>& coefs, size_t l_max,
+    size_t m_max) {
+  SpherepackIterator iter(l_max, m_max);
+  auto spherepack_coefs =
+      make_with_value<DataVector>(iter.spherepack_array_size(), 0.);
+
+  const double sqrt_2_by_pi = sqrt(2. / M_PI);
+  for (size_t l = 0; l <= l_max; ++l) {
+    iter.set(l, 0);
+    spherepack_coefs[iter()] =
+        sqrt_2_by_pi * std::real(gsl::at(gsl::at(coefs, l), 0));
+    for (size_t m = 1; m <= std::min(l, m_max); ++m) {
+      iter.set(l, m);
+      spherepack_coefs[iter()] =
+          pow(-1, m) * sqrt_2_by_pi * std::real(gsl::at(gsl::at(coefs, l), m));
+      iter.set(l, -m);
+      spherepack_coefs[iter()] =
+          pow(-1, m) * sqrt_2_by_pi * std::imag(gsl::at(gsl::at(coefs, l), m));
+    }
+  }
+  return spherepack_coefs;
+}
+
+// Generates the map, time, and a FunctionOfTime
+template <typename TransitionFunc>
+void generate_random_map_time_and_f_of_time(
+    const gsl::not_null<CoordinateMaps::TimeDependent::Shape<TransitionFunc>*>
+        map,
+    const gsl::not_null<double*> time,
+    const gsl::not_null<FunctionsOfTimeMap*> functions_of_time, size_t l_max,
+    size_t m_max, const std::array<double, 3>& center,
+    const TransitionFunc& transition_func, const DataVector& ylm_coefs,
+    gsl::not_null<std::mt19937*> generator) {
+  std::string f_of_t_name{"test_coefs"};
+
+  *map = CoordinateMaps::TimeDependent::Shape<TransitionFunc>(
+      center, l_max, m_max, transition_func, f_of_t_name);
+  // Choose a random time for evaluating the FunctionOfTime
+  std::uniform_real_distribution<double> time_dis{-1.0, 1.0};
+  *time = time_dis(*generator);
+  std::uniform_real_distribution<double> coef_dis{0.2, 2.};
+
+  // Set derivatives to zero
+  const auto dtcoefs = DataVector(ylm_coefs.size(), 0.);
+  const auto ddtcoefs = DataVector(ylm_coefs.size(), 0.);
```

Comment:
```suggestion
  const DataVector ddtcoefs{ylm_coefs.size(), 0.};
```

(nitpicking, more below)